RecoCTPPS/TotemRPLocal/plugins/TotemRPUVPatternFinder.cc

```diff
@@ -188,10 +188,11 @@ void TotemRPUVPatternFinder::produce(edm::Event& event, const edm::EventSetup& e
 
   for (auto &ids : *input)
   {
-    unsigned int detId = TotemRPDetId::rawToDecId(ids.detId());
-    unsigned int rpId = TotemRPDetId::rpOfDet(detId);
-    unsigned int plane = detId % 10;
-    bool uDir = TotemRPDetId::isStripsCoordinateUDirection(detId);
+    TotemRPDetId detId(ids.detId());
+    unsigned int plane = detId.plane();
+    bool uDir = detId.isStripsCoordinateUDirection();
+
+    TotemRPDetId rpId = detId.getRPId();
```

Comment:
@davidlange6 True, "RP" can have different meanings. Sometimes "RP" refers to the entire system of detectors, sometimes (like here) it refers to a single Roman Pot (movable container for sensors). That is one element in the detector hierarchy: arm (LHC sector), station (group of pots), a given pot, ... A part of the hierarchy is common for all CTPPS detectors, which are just different sensors inserted in the same pots. That is why this part of the hierarchy is defined in `CTPPSDetId`. Moreover, the CTPPS DetId classes are capable of describing any element of the hierarchy. So the line 195 is just asking what is the id of the pot in which is the sensor with id `detId`.